### ISSUE TITLE
disable everything related to failure when failure_transport not configured in mesenger

### ISSUE THREAD

**Issue by nikophil** [SymfonyCasts/messenger-monitor-bundle#18]
(not sure about this one...)

**Comment by nikophil** [comment]
WDYT @weaverryan ?
by "disabling" i mean "remove services from container"
but it is a little bit tricky, because we should also disable some routes, thus we have to create a custom routing loader i guess

**Comment by weaverryan** [comment]
Indeed. If we disable things based on the configuration of a *different* service, it would need to be done in a compiler pass. But it *is* a nice thing to do. So:

a) Probably let's not worry about routes. But I guess we could at least write those controllers so that if you try to access them, they give you a good error about needing the failure transport to be configured.

b) Removing the services *would* be nice, in a compiler pass. And, we may (?) not need to remove many. What I mean is, we would just need to remove any event listeners and any service we use directly in those failure controllers. If there are some additional, lower-level services that these other services depend on, we don't really need to remove those, as the container would remove them automatically because we removed all services using them, if that makes sene :p

**Comment by nikophil** [comment]
but the problem is: if i remove from the container the services which are injected in the controller, i may have an error when building the container, won't I ?

**Comment by weaverryan** [comment]
Yep, you'll need to make those services optional - so with the `on-invalid-null` and `= null` on the arguments to the constructor. But you could then immediately (in the constructor) throw an exception of the injected services are null.

It's not super clean... but it *would* be nice to clean things up.

**Comment by nikophil** [comment]
or i create another controller which handles these routes when failure transport is not configured
(and i disable original controllers)

**Comment by nikophil** [comment]
or i create the custom route loader. I've never done that, but i'm sure that's not that complicated

**Comment by weaverryan** [comment]
That's an even cooler idea :)

**Comment by nikophil** [comment]
ok, i agree, that's better if the user can see easily the routes we're adding; go for this solution!

**Comment by nikophil** [comment]
btw, you can add a prefix even with a custom route loader
for example: https://github.com/symfony/recipes/blob/master/api-platform/core/2.1/config/routes/api_platform.yaml
